palantir-java-format/src/main/java/com/palantir/javaformat/java/JavaInputAstVisitor.java

```diff
@@ -3444,10 +3465,15 @@ int declareOne(
                     initializer.get().accept(this, null);
                     builder.close();
                 } else {
-                    builder.open(
-                            Indent.If.make(typeBreak, plusFour, ZERO),
-                            BreakBehaviours.breakOnlyIfInnerLevelsThenFitOnOneLine(true),
-                            LastLevelBreakability.ABORT);
+                    if (builder.peekToken().get().equals("switch")) {
+                        // TODO(fawind): Don't break switch expression assignment
+                        builder.open(Indent.If.make(typeBreak, plusFour, ZERO));
```

Comment:
I tried deleting this block (leaving it as it was before) and it seemed like all the `FormatterIntegrationTest` cases passed.  Is it really necessary? if so can we add the integration test to capture it?

Reply:
Added a test case for that.

When using a switch-expression as an assignment and this breaking behavior:
```
builder.open(
    Indent.If.make(typeBreak, plusFour, ZERO),
    BreakBehaviours.breakOnlyIfInnerLevelsThenFitOnOneLine(true),
    LastLevelBreakability.ABORT);
```

the formatter tries to inline the last line of the statement, e.g.:

```java
int assignment(int j) {
        boolean val = switch (j) {
            case 0 -> true;
            default -> false;};    // <---
        return val;
}
```